### ISSUE TITLE
Add a menu option to set the UART baud rate. Especially, when LUA is selected for serial port.

### ISSUE THREAD

**Issue by rotorman** [opentx/opentx#8223]
**Describe the solution you'd like**
(With RadioMaster TX16S) under menu SYStem -> Hardware -> Serial port (TTL) or Serial port 2 (TTL) can be assigned to LUA.

If LUA is selected, it would be nice if there would be an additional option to select/set the baud rate for the communication.

![TX16S UARTs](https://static.rcgroups.net/forums/attachments/8/6/3/7/9/9/a14479967-209-88459210-a4c8e400-ce61-11ea-8524-e6357febdafd.png)

**Describe alternatives you've considered**
Alternatively, there could be a LUA command to set the baud rate, e.g. setSerialBaud();

**Comment by rotorman** [comment]
It looks like @3djc just [implemented](https://github.com/opentx/opentx/commit/c38e513c7e3b8cf33d2c48531f6a491ea9aa065a#diff-aa3a05d9d57c08d694cf230a36386221f56890092e54cf8d6b04fab1b2cc0702) the command. Thanks

**Comment by 3djc** [comment]
Please confirm it's working for you in tomorrow's nightly (tested here, seems ok)

**Comment by rotorman** [comment]
Huh, that's unfortunately too much asked - I am not yet setup to compile OpenTX on my own :(

**Comment by kilrah** [comment]
You don't need to, you download the automated nightly build.

**Comment by rotorman** [comment]
Thank you for your feedback!

I went ahead and installed today's nightly on my TX16S (opentx-tx16s-lua-noheli-ppmus-en-2.3.12N419.bin) and created following script that I assigned in model settings to momentary SH for test purposes:
```
local function run()
    setSerialBaudrate(57600)
    serialWrite(getValue('thr')..'\n')
end

return {run=run}
```
The good news first - I hooked the AUX1 of TX16S up to a logic analyzer and observed the output - I do get nicely 57600 baud output as expected.

The bad news - after every power-on of TX16S it does not work before I do the following: in SYSTEM -> HARDWARE -> Serial port (TTL) select OFF and then select back to LUA again.

Am I doing something wrong or does the fw need a fix?

**Comment by 3djc** [comment]
Try with AUX2 please

**Comment by rotorman** [comment]
That was a good tip - indeed with AUX2 I do NOT need to reset the Serial Port setting under SYSTEM->HARDWARE.
Thus the issue I posted 2 hours ago is only with AUX1.

But I did notice sometimes a spurious peak, please see the logic analyzer screenshot (with AUX2):
![Spurious](https://user-images.githubusercontent.com/21011587/105179686-e80ec180-5b29-11eb-85ba-aaec9e53c7fb.png)

It can be seen on TX (Channel 0 of logic analyzer) and also on RX (Channel 1) lines (the 0x2D 0x32 0x0A are expected, the first peak not). Maybe it stems from the setSerialBaudrate() call.  If you look in the decoded protocol window bottom right, then you see 0xFF is sometimes there, but most of the time not.

One more thing - I tried asking RadioMaster support, but did not get an answer - is the AUX 2 CTR pin connected at all or implemented in OpenTX? And for hardware flow, CTS/RTS is typically used, but what function does "CTR" have? Thanks!

**Comment by 3djc** [comment]
There is definitively a very weird issue, investigating

CTR is not currently used or managed by OTX.

**Comment by 3djc** [comment]
@rotorman ca you test tomorow ? It should be fixed. As a side comment, you should not send you setSerialBaudrate() everytime you wrote something if you make a more complex script

**Comment by rotorman** [comment]
Yes, eager to. Tomorrow I will get the next nightly and flash it and try again.

About the script, yes - this was just for testing, But I changed now also for testing it to:

```
local function init_func()
	setSerialBaudrate(57600)
end

local function run_func()
	serialWrite(getValue('thr')..'\n')
end

return { run=run_func, init=init_func }
```

**Comment by rotorman** [comment]
I flashed opentx-tx16s-lua-noheli-ppmus-en-2.3.12N420.bin and can confirm that now both serial ports of TX16S (AUX and AUX2) work (also without glitches on RX and TX lines). Awesome, thank you 3djc!

My dream of being able to skip extra hardware (Mav2PT Teensy board) to get MAVLink telemetry directly into Yaapu telemetry screens has come one step closer.

I wonder what is the reason to have under SYSTEM -> HARDWARE the Max bauds setting at all? What does 400000 or 115200 change? I wonder if this setting is now necessary at all and could be removed from menu for clarity?

*****
There is presently a problem with EX1 and EX2 analog inputs, at least with nightlies N419 and N420 that I tested. When flashed back to v.2.3.11 release version, the analog input bug is away. I'll open a new issue for that.
